src/lib/MovingAverage.java

```diff
@@ -10,7 +10,10 @@
 	 * Creates the average 
 	 * @param size  used to create the average   
 	 */
-	public MovingAverage(int windowSize){
+	public MovingAverage(int windowSize) throws IllegalArgumentException{
```

Comment:
Get rid of the throws clause.  `IllegalArgumentException` is not checked.  Read up on the difference